photoslibraryapi/src/main/java/com/google/photos/library/v1/PhotosLibraryClient.java

```diff
@@ -319,4 +319,15 @@ public final ListMediaItemsPagedResponse listMediaItems() {
   public final ListSharedAlbumsPagedResponse listSharedAlbums() {
     return super.listSharedAlbums(false);
   }
+
+  @Override
+  public void close() {
+    super.close();
+    try {
+      uploadStub.close();
+    } catch (Exception e) {
```

Comment:
While it is true that `BackgroundResourceAggregation.shutdown()` doesn't throw checked exceptions, it could throw `RuntimeException` ones though. I could maybe filter those ones and re-throw them without the need to change the `InternalPhotosLibraryClient.close()` signature. Also, I don’t think the removal of the `throw Exception` in  `BackgroundResource.close()` will happen any time soon since there hasn’t been a new major version upgrade in 3 years. I suppose it's quite unlikely they are going to introduce a checked exception for  `BackgroundResourceAggregation.shutdown()`, but if that happens this change will silently ignore it. A workaround for both issues that allows to capture any Exception without the need to modify the signature could be to do just this:
```java
try {
￼    uploadStub.close();
￼} catch (Exception e) {
   throw new RuntimeException(e);
}
```
What's your opinion on this @jfschmakeit ?

Reply:
Yes, this is an interesting one. (There does seem to be a bit of a mismatch further in the gapic/autogenerated side.)

I think we should not silently catch exceptions, even if they are unlikely to occur. FWIW, if the checked exception is ever removed from `BackgroundResource.close()`, this would come up as a major breaking change that we can handle at the same time.

I think the cleanest way to handle this is actually your final proposal - leaving the method signature as-is and rethrowing the exception as a `RuntimeException`. This follows the spirit of the underlying exception that would be thrown from `uploadStub.close()` and is also similar to the way this is handled by the `InternalPhotosLibraryClient`. Given there are no guarantees on the type of Exception, this might be the best approach.